pkg/controller/drain.go

```diff
@@ -414,6 +414,7 @@ func (o *DrainOptions) getGlobalTimeoutForPodsWithoutPV(pods []*api.Pod) time.Du
 
 func (o *DrainOptions) evictPods(attemptEvict bool, pods []api.Pod, policyGroupVersion string, getPodFn func(namespace, name string) (*api.Pod, error)) error {
 	returnCh := make(chan error, len(pods))
+	defer close(returnCh)
```

Comment:
Yes, the closing of this channel is not handled properly in the go routines that follow. These go routines don't check whether the caller has/has not closed the channel before writing to them. However, this [`for`](https://github.com/gardener/machine-controller-manager/pull/341/files/735d9736cbfcbc42b67d4a4bd69c70e36c9f14bd#diff-face99aebd3cd79577003f7df747c863R443-R449) loop makes sure that all go routines have already returned, after which the channel is closed. So we are safe.
This could be a potential issue if those go routines themselves have a bug.

Reply:
So I shall keep it as it is for now.